eng/MSBuild/Packaging.props

```diff
@@ -4,7 +4,7 @@
     <SignArtifacts>$(SignAssembly)</SignArtifacts>
 
     <!-- Arcade SDK marks all but *.pkgproj as non-packable, we need to override. -->
```

Comment:
This comment should be updated as dotnet/extensions doesn't use pkgprojs. cc @RussKie

Reply:
Could you please help me understand why it needs an update? The comment states that all projects except *.pkgproj are non-packable by default.

Reply:
This comment comes from dotnet/runtime which used pkgprojs. This repositories doesn't use pkgprojs so it doesn't make sense to mention them in this repository at all.

Reply:
Per my other comment I think you should just keep arcade's default of `false` here and have packages opt-in by setting it themselves, or setting `Stage`.

Reply:
> This comment comes from dotnet/runtime which used pkgprojs. This repositories doesn't use pkgprojs so it doesn't make sense to mention them in this repository at all.

Nope :) This comment reflects the [Arcade's configuration](https://github.com/dotnet/arcade/blob/5cd8dc258e8177b8542adf7abe45e7cfc1c1b1c0/src/Microsoft.DotNet.Arcade.Sdk/tools/ProjectDefaults.props#L25-L26). 
![image](https://github.com/dotnet/extensions/assets/4403806/0da9f779-9f37-409e-b5c4-bd2a37a5f21a)

I added it on purpose to help with questions like https://github.com/dotnet/extensions/pull/4254#discussion_r1287757994.